frontend/src/email/EmailForm.jsx

```diff
@@ -261,6 +322,37 @@ class EmailForm extends Component {
     });
   }
 
+  async fetchStatus(messageId) {
+    const response = await fetch(`${MSG_SERVICE_PATH}/email/${messageId}/status`);
+    if (!response.ok) {
+      throw Error('Could not connect to Showcase Messaging API for health check: ' + response.statusText);
+    }
+    return await response.json().catch(error => {
+      throw Error(error.message);
+    });
+  }
+
+  async asyncForEach(array, callback) {
+    for (let index = 0; index < array.length; index++) {
+      await callback(array[index], index, array);
+    }
+  }
```

Comment:
If you don't actually require sequential looping, running a `Promise.all([array of async functions])` and then awaiting on that promise could be more efficient.

Reply:
I wanted it sequentially because the order the messages/statuses in the list is important.